tianshou/data/batch.py

```diff
@@ -268,6 +268,9 @@ def __repr__(self) -> str:
     def __iter__(self) -> Iterator[Self]:
         ...
 
+    def __eq__(self, other: Self) -> bool:  # type: ignore
```

Comment:
why type ignore?

The second argument should be of type Any, and in the implementation you should do:

```
if not isinstance(other, self.__class__):
    return False
```

before any other logic. You can't restrict what a user will put on the other side of  `==`

Reply:
I ignored it because the current typing is in conflict with the super-class type that allows `other: object`.

You are right, we cannot control what the user will use for equality, so we should just make the check above right at the beginning.